src/batchnorm/problem_description.cpp

```diff
@@ -0,0 +1,183 @@
+/*******************************************************************************
```

Comment:
We need to run batchnorm tests for correctness and performance with resnet and other network configs.  Are those tests done?  Also, we need to see any gaps in testing as changes are many..

Reply:
I have used a simple self-made test set + corner cases for different condition paths. My script multiplies parameter value sets, limits them by image size, and dumps a lot of data from logs like times, solver name and compilation details. That data is later automatically compared with similar data gathered from the develop. I would be happy to execute whatever to test if you would like.

<details>
  <summary>Batchnorm parameter values used in my core bench</summary>

  ```powershell
$DataTypes = @("", "fp16")
$Directions = @(0, 1, 2)
$Modes = @(0, 1)
$InputSizes = @(@(64, 128), @(64, 128), @(256, 256), @(720, 480), @(1280, 760), @(1920, 1080), @(3840, 2160))
$Channels = @(1, 16, 32, 50, 64, 128)
$BatchSizes = @(1, 10, 50, 100)
$backwardsPropation = @($false, $true)
$run = @($false, $true)
$save = @($false, $true)
$Iterations = 10
$Limit = 32Mb
# This limits data_type_size*W*H*batchsize*in_channels to make bench time sane
# 32Mb was used here because as with larger images multi-kernel paths gets invoked a lot, and corner cases for that were hand-picked later.
  ```

</details>

If you would like more details on the test script I could post full files somewhere.

Reply:
Here are some configs for accuracy and performance tests. Can you please try with various batch sizes(n =32, 64,...1024)?  

FP32:
#!/bin/bash

./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 2048 -H 7 -W 7 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 1024 -H 14 -W 14 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 512 -H 7 -W 7 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 256 -H 14 -W 14 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 256 -H 56 -W 56 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 64 -H 112 -W 112 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 64 -H 56 -W 56 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 512 -H 28 -W 28 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnorm -m 1 -s 1 -r 1 -n ${1} -c 128 -H 28 -W 28 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0

FP16:
#!/bin/bash

./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 2048 -H 7 -W 7 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 1024 -H 14 -W 14 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 512 -H 7 -W 7 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 256 -H 14 -W 14 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 64 -H 112 -W 112 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 256 -H 56 -W 56 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 64 -H 56 -W 56 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 512 -H 28 -W 28 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0
./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -r 1 -n ${1} -c 128 -H 28 -W 28 -V ${3} -F 1 -b 0 -i ${2} -t 1 -w 0

Reply:
In progress

Reply:
No significant difference

Reply:
I can post the full comparison if you want.

Reply:
Good.